DQM/DTMonitorModule/src/DTLocalTriggerSynchTask.cc

```diff
@@ -40,7 +41,8 @@ using namespace std;
 DTLocalTriggerSynchTask::DTLocalTriggerSynchTask(const edm::ParameterSet& ps)
     : nevents(0),
       tTrigSync{DTTTrigSyncFactory::get()->create(ps.getParameter<std::string>("tTrigMode"),
-                                                  ps.getParameter<edm::ParameterSet>("tTrigModeConfig"))},
+                                                  ps.getParameter<edm::ParameterSet>("tTrigModeConfig"),
+                                                  consumesCollector())},
```

Comment:
After the last migration of this file to esConsumes in:
https://github.com/cms-sw/cmssw/commit/a1bda9d3e446e6481f5ed67dacb768d4470a7945#diff-d9480f5b2556d32e66f2d2de9ec484f1f221143fdb7c121054422dd9d365141c
the class was working without complains by cmsbuild.

For my understanding, could you explain me why a consumesCollector is needed in this case? It is not clear from the explanation in:
https://twiki.cern.ch/twiki/bin/view/CMSPublic/SWGuideHowToGetDataFromES#In_ED_module
Thanks

Reply:
@jfernan2 for the moment, we support 'partial migrations' to esConsumes. That is a module (and its helpers) can use a mix of EventSetup `get` calls using and not using `ESGetToken`. That is why the previous change worked. This change deals with what was missed. You can see from here

https://github.com/cms-sw/cmssw/blob/d69ec70ce6481bc54c3c8277f90fede68459abb2/CalibMuon/DTDigiSync/plugins/DTTTrigSyncFromDB.cc#L144-L145

that this plugin loaded by that factory does access the EventSetup and therefore requires the change.